doc/pages/expansions.asciidoc

```diff
@@ -1,199 +1,294 @@
 = Expansions
 
-== Strings
+While parsing a command (see <<command-parsing#,`:doc command-parsing`>>),
+Kakoune recognises certain patterns and will replace them with their
+associated value before executing the command. These patterns are called
+expansions.
 
-\'strings'::
-    uninterpreted strings, use a backslash (\') to escape the separator
+Every expansion consists of a `%`, followed by the expansion _type_ (one
+or more alphabetic characters), a nestable punctuation character (`(`, `[`,
+`{`, or `<`), and then all the text up to and including its matching opposite
+(`)`, `]`, `}`, or `>`). It doesn't matter which character is used, but
+`{}` are most common. For example, the command `echo %val{session}` will
+echo the ID of the current Kakoune session.
 
-"strings"::
-    expanded strings, % strings (c.f. next section) contained are expended,
-    use a backslash (\\%) to escape the separator
+Expansions are processed when unquoted and anywhere inside double-quoted or
+%-strings, but not inside unquoted words, inside single-quoted strings, or
```

Comment:
Thats incorrect, %-strings do not expands nested % strings.

Reply:
Oops! Was that ever true, or am I just confused?

Reply:
They never did.